include/rxmesh/kernels/attribute.cuh

```diff
@@ -94,6 +94,79 @@ __launch_bounds__(blockSize) __global__
     }
 }
 
+struct CustomMaxPair
+{
+    template <typename T>
+    __device__ __forceinline__ cub::KeyValuePair<int, T> operator()(
+        const cub::KeyValuePair<int, T>& a,
+        const cub::KeyValuePair<int, T>& b) const
+    {
+        return (b.value > a.value) ? b : a;
+    }
+};
+
+struct CustomMinPair
+{
+    template <typename T>
+    __device__ __forceinline__ cub::KeyValuePair<int, T> operator()(
+        const cub::KeyValuePair<int, T>& a,
+        const cub::KeyValuePair<int, T>& b) const
+    {
+        return (b.value < a.value) ? b : a;
+    }
+};
+template <class T, uint32_t blockSize, typename HandleT, typename Operation>
+__launch_bounds__(blockSize) __global__
+    void arg_max_kernel(const Attribute<T, HandleT> X,
+                    uint32_t                    attribute_id,
+                    Operation                   op, //can be either max or min operation
+                    const uint32_t              num_patches,
+                    const uint32_t              num_attributes,
+                    cub::KeyValuePair<HandleT, T>*  d_block_output,
+                    uint32_t                    attribute_id)
+{
+    using LocalT = typename HandleT::LocalT;
+
+    assert(X.get_num_attributes() == 1); //we can only take arg max for a scalar attribute
+
+    uint32_t p_id = blockIdx.x;
+    if (p_id < num_patches) {
+        const uint16_t element_per_patch = X.size(p_id);
+        cub::KeyValuePair<HandleT, T> thread_val;
+        thread_val.value = std::numeric_limits<T>::lowest();
+        thread_val.key   = 0;
+        for (uint16_t i = threadIdx.x; i < element_per_patch; i += blockSize) {
+
+            if (X.get_patch_info(p_id).is_owned(LocalT(i)) &&
+                !X.get_patch_info(p_id).is_deleted(LocalT(i))) {
+
+                if (attribute_id != INVALID32 ) 
+                {
+                    HandleT handle(p_id, i);
+                    cub::KeyValuePair<HandleT, T> current_pair(handle, X(p_id, i, attribute_id));
+                    thread_val = op(thread_val, current_pair);
+                }
+                else {
+                    for (uint32_t j = 0; j < num_attributes; ++j) 
+                    {
+                        HandleT                       handle(p_id, i);
+                        cub::KeyValuePair<HandleT, T> current_pair(handle, X(p_id, i, j));
+                        thread_val = op(thread_val, current_pair);
+                    }
+                }
+            }
+        }
+        typedef cub::BlockReduce<cub::KeyValuePair<HandleT, T>, blockSize> BlockReduce;
+        __shared__ typename BlockReduce::TempStorage temp_storage;
+        cub::KeyValuePair<int, T> block_aggregate = BlockReduce(temp_storage).Reduce(thread_val, op);
```

Comment:
Should not the `block_aggregate` be the same type as the input, i.e., `cub::KeyValuePair<HandleT, T>`?